src/hooks/api/useCashFlow.ts

```diff
@@ -4,44 +4,71 @@ import type { UseQueryResult } from '@tanstack/react-query';
  *
- * Returns an array containing the account guid, the total for that account, the type and
- * the name.
+ * The query works for transactions in multiple splits and in different currencies.
+ * We do that by calculating the exchange rate in the transaction and applying it to
+ * the splits for the other accounts.
  */
 export function useCashFlow(
   account: string,
   selectedInterval?: Interval,
-): UseQueryResult<{ guid: string, total: number, type: string, name: string }[]> {
+): UseQueryResult<CashFlowRow[]> {
   const { data: defaultInterval } = useInterval();
   const interval = selectedInterval || defaultInterval;
 
   const result = useQuery({
     queryKey: [...Split.CACHE_KEY, account, 'cashflow', interval.toISODate()],
-    queryFn: () => Split.query(`
-      SELECT
-        splits.account_guid as guid,
-        accounts.name,
-        accounts.account_type as type,
-        SUM(cast(splits.quantity_num as REAL) / splits.quantity_denom) as total
-      FROM transactions AS tx
-      JOIN splits ON splits.tx_guid = tx.guid
-      JOIN accounts ON splits.account_guid = accounts.guid
-      WHERE tx.guid IN (
-        SELECT DISTINCT tx_guid
-        FROM splits
-        WHERE account_guid = '${account}'
-      )
-      AND tx.post_date >= '${((interval as Interval).start as DateTime).toSQLDate()}'
-      AND tx.post_date <= '${((interval as Interval).end as DateTime).toSQLDate()}'
-      GROUP BY splits.account_guid
-      HAVING SUM(cast(splits.quantity_num as REAL) / splits.quantity_denom) != 0
-      ORDER BY total
-    `),
+    queryFn: async () => {
+      const rows = await Split.query(`
+        SELECT
+          splits.account_guid as guid,
+          accounts.name as name,
+          accounts.account_type as type,
+          (
+            SELECT commodities.mnemonic 
+            FROM commodities
+            JOIN accounts AS main_account ON main_account.guid = '${account}'
+            WHERE commodities.guid = main_account.commodity_guid
+          ) AS mnemonic,
+          SUM(((cast(main_split.quantity_num as REAL) / main_split.quantity_denom) / (cast(main_split.value_num as REAL) / main_split.value_denom)) * (cast(splits.value_num as REAL) / splits.value_denom)) as total
+
+        FROM transactions AS tx
+        JOIN splits ON splits.tx_guid = tx.guid
+        JOIN accounts ON splits.account_guid = accounts.guid
+        JOIN splits AS main_split ON main_split.tx_guid = tx.guid AND main_split.account_guid = '${account}'
+
+        WHERE tx.guid IN (
+          SELECT DISTINCT tx_guid
+          FROM splits
+          WHERE account_guid = '${account}'
+        )
+        AND accounts.guid != '${account}'
+        AND tx.post_date >= '${((interval as Interval).start as DateTime).toSQLDate()}'
+        AND tx.post_date <= '${((interval as Interval).end as DateTime).toSQLDate()}'
+
+        GROUP BY splits.account_guid, accounts.name, accounts.account_type, mnemonic
```

Comment:
well this query took quite a while to put together... basically the strategy it follows is:

1. grab each split and tie it to each account except for the account we are passing
2. grab the main split of each transaction
3. calculate the total for each split as `main_split.quantity / main_split.value * split.value`. This gives the total for that split **in the main account's currency**. This is what we want for the cash flow report.
4. hardcode the commodity to the commodity of the account we are generating the cash flow for.